src/network/networkd-state-file.c

```diff
@@ -133,6 +173,15 @@ static int link_put_dns(Link *link, OrderedSet **s) {
                 }
         }
 
+        if (network_ndisc_use_dnr(link->network)) {
+                NDiscDNR *a;
+
+                SET_FOREACH(a, link->ndisc_dnr) {
+                        r = ordered_set_put_in6_addrv(s, (struct in6_addr *)a->resolver.addrs,
+                                        a->resolver.n_addrs);
+                }
```

Comment:
I'm not sure where the ADNs go. It adds IP addresses only as far as I can see. As far as I understand `ordered_set_put_dns_servers` should be called somewhere instead.

Reply:
Hm yeah. I forgot to inquire about this, but I was also a little confused by the `&a->router` in the dns section just above, looks wrong. I'll correct both to something sensible.

Reply:
> &a->router in the dns section just above, looks wrong

Looking at https://github.com/systemd/systemd/pull/30952/commits/fc428d0bb23e49e5b5ada5ada3b218992bc63d51 it seems that that bug fix should go to a separate PR. (I'm not sure why the CI didn't catch it. `radvd` supports `RDNSS` as far as I can remember)

Reply:
Ok.